### PR TITLE
fix: use sbom uid instead of root package purl in product-by-cve query

### PR DESCRIPTION
**Pull request by dejanb**

### CODE REVIEW

File: spog/api/src/service/guac.rs [hunk 1]
```diff
@@ -168,35 +178,53 @@ impl GuacService {
...
     pub async fn product_by_cve(&self, id: String) -> Result<CveDetails, Error> {
         let result = self.client.semantic().product_by_cve(&id).await?;
         let mut products = BTreeMap::<ProductCveStatus, Vec<ProductRelatedToCve>>::new();
 
         for product in result {
-            let id = product.root.try_as_purls()?[0].name().to_string();
+            let root = self.get_purl(product.root)?.clone();
+
+            let sbom = self.client.intrinsic().has_sbom(&root.into()).await?;
+            let uid = &sbom[0].uri;
```
Comment (ctron) [inline spog/api/src/service/guac.rs:200]: I think this could again panic.

**Review by ctron — CHANGES_REQUESTED** [review]
Looks good. I think we should try to get rid of that remaining panic.

File: spog/api/src/service/guac.rs (hunk 1 above)
Comment (dejanb) [reply spog/api/src/service/guac.rs:200]: Right. Forgot about that one :)

Comment (dejanb) [reply spog/api/src/service/guac.rs:203]: Fixed

**Review by ctron — APPROVED** [review]

**Pull request merged**